README.md

```diff
@@ -92,5 +93,34 @@ var client = new ClientBuilder()
 
 Great! Now you have Orleans configured to use Azure CosmosDB as the back storage for its providers!
 
+## Custom partition keys
+By default the grain type is used as partition key, however this limits the storage size for a single grain type as a single logical partition has an upper limit of 10 GB storage.
+
+It is possible to override the default by implementing a custom IPartitionKeyProvider. The custom implementation must be provided to the `CosmosDBStorageOptions` configuration using the `PartitionKeyProvider` property.
+
+If no IPartitionKeyProvider is used, the full name of the grain type will be used as partition key. 
+
+***Example***
+```cs
+public class PrimaryKeyPartitionKeyProvider : IPartitionKeyProvider
+{
+    public string GetPartitionKey(string grainType, GrainReference grainReference)
```

Comment:
Please update the sample with your latest changes and `ValueTask<T>`.

Reply:
Good catch